lib/ex_machina/ecto.ex

```diff
@@ -191,7 +192,30 @@ defmodule ExMachina.Ecto do
     end
   end
 
-  defp set_persisted_belongs_to_ids(record = %{__struct__: struct, __meta__: %{__struct__: Ecto.Schema.Metadata}}) do
+  defp handle_embeds(record = %{__struct__: struct}) do
+    Enum.reduce(struct.__schema__(:embeds), record, fn(embed_name, record) ->
+      record
+      |> Map.get(embed_name)
+      |> handle_embed(record, embed_name)
+    end)
+  end
+
+  defp handle_embed(original_embed, record, embed_name) do
+    case original_embed do
+      %{__struct__: _} ->
+        embed = recursively_strip(original_embed)
+        Map.put(record, embed_name, embed)
+
+      list when is_list(list) ->
+        embeds_many = Enum.map(original_embed, &recursively_strip/1)
+        Map.put(record, embed_name, embeds_many)
+
```

Comment:
Could you please remove the newline here and on line 208 to keep with existing style?